Approved/implicitly-scoped-qubit-allocation.md

```diff
@@ -0,0 +1,314 @@
+---
+title: Implicitly-scoped qubit allocation
+description: Provides more convenient syntax for allocating qubits.
+author: Sarah Marshall
+date: TBD
+---
+
+# Proposal
+
+1. Rename `using` and `borrowing` keywords to `use` and `borrow`.
+2. Allow `use` and `borrow` statements to be followed by either a block `{ ... }` or a semicolon `;`.
+3. Remove the requirement for parentheses around headers for all block statements.
+
+# Justification
+
+Qubit allocation is currently awkward in Q# for two reasons:
+
+1. It requires creating a new block to define the lifetime of the qubit, even when the intended lifetime is the same as the current block.
+2. Allocating multiple qubits at the same time requires either tuple destructuring or nested blocks, both of which can be hard to read.
+
+As a quantum programming language, qubit allocation is at the core of Q#.
+Both the simple case of allocating a single qubit for the remainder of the current block, and more complicated scenarios involving multiple qubit variables that are both clean and borrowed and have varying lifetimes, should have clear, intuitive syntax that doesn't require boilerplate.
+
+# Description
+
+Qubits are a managed resource in Q#: when qubits are allocated, they are valid only within the new block declared at allocation, after which they are released and can no longer be used.
+This proposal makes qubit allocation in Q# always scoped to the current block.
```

Comment:
I believe always is incorrect here; the proposal introduces the option to allocate qubits that are scoped to the current block.